README
Version 1, portage LaTeX par Antoine Luboz (luboz@ipno.in2p3.fr)
Mise en page inspirée des documents PV et procédure d'Atttrium.

INTRODUCTION
    Le projet permet la rédaction de comptes-rendus, rapports techniques, procès-verbaux, etc. en utilisant la mise en page caractéristique et propre des documents rédigés en LaTeX.
    Un très bon cours sur les bases du LaTeX est disponible sur Open Classrooms  https://openclassrooms.com/courses/redigez-des-documents-de-qualite-avec-latex
    De plus, LaTeX dispose d'une communauté très active et la plupart des problèmes ou questions trouve une réponse très rapide avec une simple recherche Google.

    Voici ci-après quelques recommandations pour que la compilation du document se passe au mieux.

STRUCTURE DU PROJET
    FICHIERS PROJET LATEX, dossier Files
        Les fichiers nécessaires à la bonne compilation du projet sont situés dans le dossier files. Il est normalement pas nécessaire d'y toucher mais à titre informatif ou au vu de futures modifications. Chacun est libre de modifier la structure du projet, merci de maintenir le changelog à jour ainsi que le readme explicatif. Voici le détail du dossier:
            Logos
                les logos des instituts sont placés dans le dossier Logos - Merci de me contacter si vous trouvez un logo vectoriel de l'IPNO (format eps, svg, pdf).
            changelog
                répertorie les changements apportés à la structure des fichiers du projet. Tout changement est à signaler dans le fichier changelog.
            commandes.tex
                répertorie les commandes utilisables (guillemets français, nombres romains...) Dans la suite du document.
            main.tex
                est le fichier compilé, les autres fichiers tex sont inclus dans celui-ci.
            packages.tex
                contient la liste d'appels de packages et leurs options. C'est ici qu'on peut également modifier les options du langage informatique (citation de code avec lstlistings), l'affichage des nombres et unités (siunitx) ou la langue du texte.
            titlepage.tex
                est la page de garde.
            unsrt-fr.bst
                est le fichier de configuration pour une bibliographie s'affichant en français.
    
    AUTRES FICHIERS DU PROJET
        Les seuls fichiers qui nécessitent une modification sont normalement ceux listés à la racine. Chacun contiennent des instructions en en-tête en commentaire LaTeX (%). Voici une liste avec description succincte.
            configuration.tex
                Fichier de configuration du document où on indique l'auteur, le titre...
            histModifications.tex
                Tableau dans lequel on indique l'historique des modifications de la lignée de fichier, rajoutez une ligne par nouveau fichier diffusé.
            resume.tex
                Résumé du fichier que l'on crée qui s'affichera sur la page de garde, restez succins.
            bibliographie.bib
                Fichier répertoriant les références bibliographiques au format BibTeX.
            contenu.tex
                Le contenu à proprement parler du document, on utilise les balises \chapter{}, \section{} et autres commandes LaTeX.

RECOMMANDATIONS, CONSEILS ET INSTRUCTIONS

    Compiler sur le fichier files/main.tex

    Utilisez \frquotes{} pour les guillemets français

    Dans la page de configuration, remplacez les accents comme suit:
        é devient \'{e}
        à devient \`{a}
        Ë devient \"{E}
    utilisez la recherche et remplacement CTRL + F, cf. documentation pour d'autres types d'accents.

    Bibliographie
        La plupart des références bibliographiques peuvent-être trouvées sur Google Scholar, Mendeley... en choisissant d'exporter en BIB.
        Une référence se présente sous cette forme:
        @type{tag, title = {titre}, author = {auteur}, year = {année}, month = mois {jan,feb...}, <d'autres paramètres spécifiques aux différents types>}
        Une référence est citée en utilisant la commande \cite{tag}, les références peuvent-être combinées \cite{tag1,tag2}

Files/changelog
09/04/2018
    Création projet par Antoine Luboz (luboz@ipno.in2p3.fr)

FichiersAnnexes/routinePython_ouvertureVanne.py
# -*- coding: utf-8 -*-
"""
Created on Wed May  2 14:10:46 2018

@author: Antoine Luboz

"""
# Se placer dans ce directory
#cd C:\Users\luboz.IPN\Documents\Projets\QualificationFour\ExperienceCapots
#%% Import des packages
import numpy as np 
import pandas as pd 
import matplotlib.pyplot as plt 
import matplotlib.ticker
#from FonctionsPython.extractionCsv import extractionCsv

from matplotlib import rc
rc('font',**{'family':'sans-serif','sans-serif':['Helvetica']})
## for Palatino and other serif fonts use:
#rc('font',**{'family':'serif','serif':['Palatino']})
rc('text', usetex=True)
#from matplotlib.ticker import MultipleLocator, FormatStrFormatter
#from matplotlib2tikz import save as tikz_save
import matplotlib as mpl
mpl.use("pgf")
pgf_with_pdflatex = {
    "pgf.texsystem": "pdflatex",
    "pgf.preamble": [
         r"\usepackage[utf8]{inputenc}",
         r"\usepackage[T1]{fontenc}",
         #r"\usepackage{cmbright}",
         ]
}
mpl.rcParams.update(pgf_with_pdflatex)
        
from matplotlib.ticker import Locator
from scipy.optimize import curve_fit

class MinorSymLogLocator(Locator):
    """
    Dynamically find minor tick positions based on the positions of
    major ticks for a symlog scaling.
    """
    def __init__(self, linthresh):
        """
        Ticks will be placed between the major ticks.
        The placement is linear for x between -linthresh and linthresh,
        otherwise its logarithmically
        """
        self.linthresh = linthresh

    def __call__(self):
        'Return the locations of the ticks'
        majorlocs = self.axis.get_majorticklocs()

        # iterate through minor locs
        minorlocs = []

        # handle the lowest part
        for i in range(1, len(majorlocs)):
            majorstep = majorlocs[i] - majorlocs[i-1]
            if abs(majorlocs[i-1] + majorstep/2) < self.linthresh:
                ndivs = 10
            else:
                ndivs = 9
            minorstep = majorstep / ndivs
            locs = np.arange(majorlocs[i-1], majorlocs[i], minorstep)[1:]
            minorlocs.extend(locs)

        return self.raise_if_exceeds(np.array(minorlocs))

    def tick_values(self, vmin, vmax):
        raise NotImplementedError('Cannot get tick locations for a '
                                  '%s type.' % type(self))
        
#classe des choix de graphes à tracer
class Plots_ouvertureVanne:
    """"Classe permettant de choisir quel graphes tracer.
    - sans_capots, logique, defaut = True
    - P_LV, logique, defaut = True
    - P_HV, logique, defaut = True
    - Subplots, logique, defaut = True"""
    
    def __init__(self,sans_capots, capots_10mm, capots_1mm):
        self.sans_capots = True
        self.capots_10mm = True
        self.capots_1mm = True

from uncertainties import ufloat#incertitudes
        
with open('Graphes/resultats_vanne.txt','w') as file:
    file.write("---------------------------------------\nOUTPUT LORS DE L'OUVERTURE DE LA VANNE")
    
#%% Parmètres de taille et constantes
largeurPlot = 6.68; #largeur en inch
ratioPlot = 1.3;
hauteurPlot = largeurPlot/ratioPlot;
volumeFour = 4500#en litre

#on défnit d'abord la fonction linéaire
def linfunc(x,a,b):
    return(a*x+b)

#%% Choix des graphes à générer (je crée cette classe pour m'entrainer aux classes)        
choix = Plots_videStatique(True,True,True)

#%% Extraction des datas
ouvertureVanne_sansCapots = pd.read_csv('Data/ouvertureVanne_sansCapots.csv',skiprows=1, infer_datetime_format = True, error_bad_lines=False)
ouvertureVanne_10mm = pd.read_csv('Data/ouvertureVanne_10mm.csv',skiprows=1, infer_datetime_format = True, error_bad_lines=False)
ouvertureVanne_1mm = pd.read_csv('Data/ouvertureVanne_1mm.csv',skiprows=1, infer_datetime_format = True, error_bad_lines=False)

#%% Suppression des colonnes inutiles et renommage
del ouvertureVanne_sansCapots["Remarques"]
del ouvertureVanne_10mm["Remarques"]
del ouvertureVanne_1mm["Remarques"]

#%% Ajout de l'erreur de chaque jauge
ouvertureVanne_sansCapots["erreur P_cav (mbar)"] = 0.4*ouvertureVanne_sansCapots["P_cav (mbar)"] ; ouvertureVanne_sansCapots["erreur P_HV"] = 0.15*ouvertureVanne_sansCapots["P_HV (mbar)"]
ouvertureVanne_10mm["erreur P_cav (mbar)"] = 0.4*ouvertureVanne_10mm["P_cav (mbar)"] ; ouvertureVanne_10mm["erreur P_HV"] = 0.15*ouvertureVanne_10mm["P_HV (mbar)"]
ouvertureVanne_1mm["erreur P_cav (mbar)"] = 0.4*ouvertureVanne_1mm["P_cav (mbar)"] ; ouvertureVanne_1mm["erreur P_HV"] = 0.15*ouvertureVanne_1mm["P_HV (mbar)"]

# Tracé des courbes
#%% P sans capots en fonction de l'ouverture de la vanne et du taux de fuite
if choix.sans_capots:

    fig = plt.figure(num='ouvertureVanne_sansCapots', figsize = (largeurPlot, hauteurPlot))
    fig, ax1 = plt.subplots()
    ax1.plot(ouvertureVanne_sansCapots["ouverture vanne (tours)"],ouvertureVanne_sansCapots['P_cav (mbar)'],'r.', label= '$P_{cav}$')
    ax1.plot(ouvertureVanne_sansCapots["ouverture vanne (tours)"],ouvertureVanne_sansCapots['P_HV (mbar)'],'b.', label= '$P_{HV}$')
    ax1.plot(ouvertureVanne_sansCapots["ouverture vanne (tours)"],ouvertureVanne_sansCapots['P_RGA (mbar)'],'k.', label= '$P_{RGA}$')
        
    plt.legend()

    plt.minorticks_on()
    plt.grid(b = True, which = 'major', axis = 'both')
    
    ax1.errorbar(ouvertureVanne_sansCapots['ouverture vanne (tours)'],ouvertureVanne_sansCapots['P_cav (mbar)'], yerr = ouvertureVanne_sansCapots['erreur P_cav (mbar)'], fmt='none', ecolor = 'k', elinewidth = 1, capsize = 1, label= '$P_{cav}$')
    ax1.errorbar(ouvertureVanne_sansCapots['ouverture vanne (tours)'],ouvertureVanne_sansCapots['P_HV (mbar)'], yerr = ouvertureVanne_sansCapots['erreur P_HV'], fmt='none', ecolor = 'k', elinewidth = 1, capsize = 1, label= '$P_{HV}$')
    #plt.errorbar(ouvertureVanne_sansCapots['Temps (min)'],ouvertureVanne_sansCapots['P_RGA (mbar)'], yerr = ouvertureVanne_sansCapots['erreur P_cav (mbar)'], fmt='none', ecolor = 'k', elinewidth = 1, capsize = 1, label= '$P_{RGA}$')

    plt.rc('text', usetex=True)
    plt.rc('font', family='serif')

    ax1.set_yscale('log', nonposy = 'mask')
    ax1.set_ylabel(r'$P$ (mbar)')
    ax1.set_xscale('linear')#symlog displays a linear interval at 0 vicinity
    ax1.set_xlabel('Ouverture de la vanne (tour)')
    #ax1.set_xlim(10.75,12.75);
    plt.minorticks_on()
    
    #ajout d'un second axe donnant le taux de fuite
    ax2 = ax1.twiny()
    ax2.set_xlim(ax1.get_xlim())
    new_tick_locations = np.array(ouvertureVanne_sansCapots["ouverture vanne (tours)"])
    new_tick_label = ['{:.2e}'.format(x) for x in ouvertureVanne_sansCapots["Taux de fuite (mbar × L/s)"]]
    ax2.set_xticks(new_tick_locations)
    ax2.set_xticklabels(new_tick_label, rotation=17)
    ax2.set_xlabel(r'Taux de fuite (mbar$\times$L/s)')
    
    #plt.title("Sans capots")
    plt.tight_layout()#pour séparer un peu les plots
    
    #export en png et pgf pour latex.
    name = "ouvertureVanne_sansCapots"
    plt.savefig(('Graphes/'+name+'.png'), format = 'png', transparent=True, dpi = 300)
    plt.savefig(('Graphes/'+name+'.pgf'), format = 'pgf')
    plt.savefig(('Graphes/'+name+'.pdf'), format = 'pdf', transparent=True)
    plt.savefig(('Graphes/'+name+'.svg'), format = 'svg', transparent=True)
    plt.show()
 
#%% Working example
#HEADERS
import numpy as np 
import pandas as pd 
import matplotlib.pyplot as plt 
import matplotlib.ticker

from matplotlib import rc
rc('font',**{'family':'sans-serif','sans-serif':['Helvetica']})
rc('text', usetex=True)

from matplotlib.ticker import Locator
from scipy.optimize import curve_fit

class MinorSymLogLocator(Locator):
    def __init__(self, linthresh):
        self.linthresh = linthresh

    def __call__(self):
        'Return the locations of the ticks'
        majorlocs = self.axis.get_majorticklocs()
        minorlocs = []
        for i in range(1, len(majorlocs)):
            majorstep = majorlocs[i] - majorlocs[i-1]
            if abs(majorlocs[i-1] + majorstep/2) < self.linthresh:
                ndivs = 10
            else:
                ndivs = 9
            minorstep = majorstep / ndivs
            locs = np.arange(majorlocs[i-1], majorlocs[i], minorstep)[1:]
            minorlocs.extend(locs)

        return self.raise_if_exceeds(np.array(minorlocs))

    def tick_values(self, vmin, vmax):
        raise NotImplementedError('Cannot get tick locations for a '
                                  '%s type.' % type(self))
	
	
#/HEADERS
	
turns = np.array([11.000, 11.500, 11.750, 12.000, 12.250, 12.375])
leak = np.array([3.89e-05, 4.63e-05, 1.67e-04, 1.45000000e-03, 8.61e-03, 1.71e-02])
pressure1 = np.array([7.9e-07, 3.0e-06, 3.5e-05, 6.1e-04, 5.1e-03, 1.8e-02])
pressure2 = np.array([8.22e-07, 8.22e-07, 8.71e-07, 1.8e-06, 1.150e-05, 7.24e-05])
pressure3 = np.array([2e-06, 2e-06, 2e-06, 1.2e-05, 1.2e-04, 6e-04])

fig = plt.figure(num='ouvertureVanne_sansCapots', figsize = (6.68, 6.68*1.3))
fig, ax1 = plt.subplots()
ax1.plot(turns, pressure1, 'r.', label= '$P_1$')
ax1.plot(turns, pressure2, 'b.', label= '$P_2$')
ax1.plot(turns, pressure3,'k.', label= '$P_3$')
     
plt.legend()

plt.minorticks_on()
plt.grid(b = True, which = 'major', axis = 'both')
 
ax1.errorbar(turns, pressure1, yerr = .4*pressure1, fmt='none', ecolor = 'k', elinewidth = 1, capsize = 1, label= '$P_{1err}$')
ax1.errorbar(turns, pressure2, yerr = .15*pressure2, fmt='none', ecolor = 'k', elinewidth = 1, capsize = 1, label= '$P_{2err}$')

plt.rc('text', usetex=True)
plt.rc('font', family='serif')

ax1.set_yscale('log', nonposy = 'mask')
ax1.set_ylabel(r'$P$')
ax1.set_xscale('linear')
ax1.set_xlabel('Opening (turns)')
#ax1.set_xlim(10.75,12.75);
plt.minorticks_on()
#plt.grid(b = True, which = 'major', axis = 'both')
    
#adding a secondary x-axis above
ax2 = ax1.twiny()
ax2.set_xlim(ax1.get_xlim())
new_tick_locations = turns
new_tick_label = ['{:.2e}'.format(x) for x in leak] #dtype?
ax2.set_xticks(new_tick_locations)
ax2.set_xticklabels(new_tick_label, rotation=17)
#ax2.xaxis.set_scientific(True)
#ax2.get_xaxis().set_major_formatter((matplotlib.ticker.Formatter(set_scientific(True)))
#ax2.get_xaxis().set_major_formatter().set_scientific(True)
ax2.set_xlabel(r'Leak rate (mbar$\times$L/s)')

plt.tight_layout()

#export png
plt.savefig(('export.png'), format = 'png', transparent=False, dpi = 300)
plt.show()

FichiersAnnexes/routinePython_descentePression.py
# -*- coding: utf-8 -*-
"""
Created on Tue Apr 24 09:55:38 2018

@author: Antoine Luboz

Liens utiles :
    https://pythonhosted.org/spyder/editor.html?highlight=cell#how-to-define-a-code-cell
    http://www.chicoree.fr/w/Fichiers_CSV_en_Python
"""
# Se placer dans ce directory
#cd C:\Users\luboz.IPN\Documents\Projets\QualificationFour\ExperienceCapots
#%% Import des packages
import numpy as np 
import pandas as pd 
import matplotlib.pyplot as plt 
#from FonctionsPython.extractionCsv import extractionCsv

from matplotlib import rc
rc('font',**{'family':'sans-serif','sans-serif':['Helvetica']})
## for Palatino and other serif fonts use:
#rc('font',**{'family':'serif','serif':['Palatino']})
rc('text', usetex=True)
#from matplotlib.ticker import MultipleLocator, FormatStrFormatter
#from matplotlib2tikz import save as tikz_save
import matplotlib as mpl
mpl.use("pgf")
pgf_with_pdflatex = {
    "pgf.texsystem": "pdflatex",
    "pgf.preamble": [
         r"\usepackage[utf8]{inputenc}",
         r"\usepackage[T1]{fontenc}",
         #r"\usepackage{cmbright}",
         ]
}
mpl.rcParams.update(pgf_with_pdflatex)

#classe des choix de graphes à tracer
class Plots_descentePression:
    """"Classe permettant de choisir quel graphes tracer.
    - P_cav, logique, defaut = True
    - P_LV, logique, defaut = True
    - P_HV, logique, defaut = True
    - Subplots, logique, defaut = True"""
    
    def __init__(self,P_cav,P_LV,P_HV,Subplots):
        self.P_cav = True
        self.P_LV = True
        self.P_HV = True
        self.Subplots = True
        
from matplotlib.ticker import Locator


class MinorSymLogLocator(Locator):
    """
    Dynamically find minor tick positions based on the positions of
    major ticks for a symlog scaling.
    """
    def __init__(self, linthresh):
        """
        Ticks will be placed between the major ticks.
        The placement is linear for x between -linthresh and linthresh,
        otherwise its logarithmically
        """
        self.linthresh = linthresh

    def __call__(self):
        'Return the locations of the ticks'
        majorlocs = self.axis.get_majorticklocs()

        # iterate through minor locs
        minorlocs = []

        # handle the lowest part
        for i in range(1, len(majorlocs)):
            majorstep = majorlocs[i] - majorlocs[i-1]
            if abs(majorlocs[i-1] + majorstep/2) < self.linthresh:
                ndivs = 10
            else:
                ndivs = 9
            minorstep = majorstep / ndivs
            locs = np.arange(majorlocs[i-1], majorlocs[i], minorstep)[1:]
            minorlocs.extend(locs)

        return self.raise_if_exceeds(np.array(minorlocs))

    def tick_values(self, vmin, vmax):
        raise NotImplementedError('Cannot get tick locations for a '
                                  '%s type.' % type(self))
        
from uncertainties import ufloat#incertitudes

#%% Parmètres de taille
largeurPlot = 6.68; #largeur en inch
ratioPlot = 1.3;
hauteurPlot = largeurPlot/ratioPlot;
#%% Choix des graphes à générer (je crée cette classe pour m'entrainer aux classes)        
choix = Plots_descentePression(True,True,True,True)

#%% Extraction des datas
descentePression_sansCapots = pd.read_csv('Data/descentePression_sansCapots.csv',skiprows=1, infer_datetime_format = True, error_bad_lines=False)
descentePression_10mm = pd.read_csv('Data/descentePression_10mm.csv',skiprows=1, infer_datetime_format = True, error_bad_lines=False)
descentePression_1mm = pd.read_csv('Data/descentePression_1mm.csv',skiprows=1, infer_datetime_format = True, error_bad_lines=False)

#%% Suppression des colonnes inutiles et renommage

del descentePression_sansCapots["Date et heure"] ; del descentePression_sansCapots["Commentaires"]
del descentePression_10mm["Date et heure"] ; del descentePression_10mm["Commentaires"]
del descentePression_1mm["Date et heure"] ; del descentePression_1mm["Commentaires"]

#%% Ajout de l'erreur de chaque jauge

descentePression_sansCapots["erreur P_cav (mbar)"] = 0.4*descentePression_sansCapots["P_cav (mbar)"] ; descentePression_sansCapots["erreur P_LV "] = 0.15*descentePression_sansCapots["P_LV (mbar)"] ; descentePression_sansCapots["erreur P_HV"] = 0.15*descentePression_sansCapots["P_HV (mbar)"]
descentePression_10mm["erreur P_cav (mbar)"] = 0.4*descentePression_10mm["P_cav (mbar)"] ; descentePression_10mm["erreur P_LV "] = 0.15*descentePression_10mm["P_LV (mbar)"] ; descentePression_10mm["erreur P_HV"] = 0.15*descentePression_10mm["P_HV (mbar)"]
descentePression_1mm["erreur P_cav (mbar)"] = 0.4*descentePression_1mm["P_cav (mbar)"] ; descentePression_1mm["erreur P_LV "] = 0.15*descentePression_1mm["P_LV (mbar)"] ; descentePression_1mm["erreur P_HV"] = 0.15*descentePression_1mm["P_HV (mbar)"]

# Tracé des courbes
#%% P_cav en fonction du temps lors du pompage
if choix.P_cav:
    plt.figure(num='pCav', figsize = (largeurPlot, hauteurPlot))
    plt.plot(descentePression_sansCapots['Temps (min)'],descentePression_sansCapots['P_cav (mbar)'],'r.', label= 'sans capots')
    plt.plot(descentePression_10mm['Temps (min)'],descentePression_10mm['P_cav (mbar)'],'b.', label= '10 mm')
    plt.plot(descentePression_1mm['Temps (min)'],descentePression_1mm['P_cav (mbar)'],'k.', label= '1 mm')
    plt.legend()
    
    plt.errorbar(descentePression_sansCapots['Temps (min)'],descentePression_sansCapots['P_cav (mbar)'], yerr = descentePression_sansCapots['erreur P_cav (mbar)'], fmt='none', ecolor = 'k', elinewidth = 1, capsize = 1, label= 'sans capots')
    plt.errorbar(descentePression_10mm['Temps (min)'],descentePression_10mm['P_cav (mbar)'], yerr = descentePression_10mm['erreur P_cav (mbar)'], fmt='none', ecolor = 'k', elinewidth = 1, capsize = 1, label= '10 mm')
    plt.errorbar(descentePression_1mm['Temps (min)'],descentePression_1mm['P_cav (mbar)'], yerr = descentePression_1mm['erreur P_cav (mbar)'], fmt='none', ecolor = 'k', elinewidth = 1, capsize = 1, label= '1 mm')

    plt.rc('text', usetex=True)
    plt.rc('font', family='serif')

    plt.yscale('log')
    #plt.ylim((1e-7,1e3))
    #yaxis = plt.gca().yaxis
    #plt.axes().yaxis.set_tick_params(which='minor', right = 'off')
    #plt.tick_params(axis='y', which='minor')
    plt.ylabel(r'$P_{cav}$ (mbar)')
    plt.xscale('symlog', linthreshy=1e-1)#symlog displays a linear interval at 0 neighborhood
    xaxis = plt.gca().xaxis
    xaxis.set_minor_locator(MinorSymLogLocator(1e-1))
    plt.xlabel('Temps (min)')
    
    #plt.xlim()
    #plt.minorticks_on()
    plt.grid(b = True, which = 'major', axis = 'both')
    
    plt.tight_layout()
    #export en png et pgf pour latex.
    name = "pompage_pCav"
    plt.savefig(('Graphes/'+name+'.png'), format = 'png', transparent=True, dpi = 460)
    plt.savefig(('Graphes/'+name+'.pgf'), format = 'pgf')
    plt.savefig(('Graphes/'+name+'.pdf'), format = 'pdf', transparent=True)
    plt.savefig(('Graphes/'+name+'.svg'), format = 'svg', transparent=True)
    plt.show()

#%% P_HV et en fonction du temps lors du pompage
if choix.P_HV:
    plt.figure(num='pHv', figsize = (largeurPlot, hauteurPlot))
    plt.plot(descentePression_sansCapots['Temps (min)'],descentePression_sansCapots['P_HV (mbar)'],'r.', label= 'sans capots')
    plt.plot(descentePression_10mm['Temps (min)'],descentePression_10mm['P_HV (mbar)'],'b.', label= '10 mm')
    plt.plot(descentePression_1mm['Temps (min)'],descentePression_1mm['P_HV (mbar)'],'k.', label= '1 mm')
    plt.legend()

    plt.errorbar(descentePression_sansCapots['Temps (min)'],descentePression_sansCapots['P_HV (mbar)'], yerr = descentePression_sansCapots['erreur P_HV'], fmt='none', ecolor = 'k', elinewidth = 1, capsize = 1, label= 'sans capots')
    plt.errorbar(descentePression_10mm['Temps (min)'],descentePression_10mm['P_HV (mbar)'], yerr = descentePression_10mm['erreur P_HV'], fmt='none', ecolor = 'k', elinewidth = 1, capsize = 1, label= '10 mm')
    plt.errorbar(descentePression_1mm['Temps (min)'],descentePression_1mm['P_HV (mbar)'], yerr = descentePression_1mm['erreur P_HV'], fmt='none', ecolor = 'k', elinewidth = 1, capsize = 1, label= '1 mm')
    
    plt.rc('text', usetex=True)
    plt.rc('font', family='serif')

    plt.yscale('log')
    plt.ylabel(r'$P_{HV}$ (mbar)')
    plt.xscale('log')
    plt.xlabel('Temps (min)')
    #plt.axes().yaxis.set_tick_params(which='minor', right = 'off')
    #plt.tick_params(axis='y', which='minor')
    #plt.xlim(); plt.ylim()
    #plt.minorticks_on()
    plt.grid(b = True, which = 'major', axis = 'both')

    #export en png et pgf pour latex.
    name = "pompage_pHv"
    plt.savefig(('Graphes/'+name+'.png'), format = 'png', transparent=True)
    plt.savefig(('Graphes/'+name+'.pgf'), format = 'pgf')
    plt.savefig(('Graphes/'+name+'.pdf'), format = 'pdf', transparent=True)
    plt.savefig(('Graphes/'+name+'.svg'), format = 'svg', transparent=True)
    plt.show()

#%% P_LV et en fonction du temps lors du pompage
if choix.P_LV:
    plt.figure(num='pLv', figsize = (largeurPlot, hauteurPlot))
    plt.plot(descentePression_sansCapots['Temps (min)'],descentePression_sansCapots['P_LV (mbar)'],'r.', label= 'sans capots')
    plt.plot(descentePression_10mm['Temps (min)'],descentePression_10mm['P_LV (mbar)'],'b.', label= '10 mm')
    plt.plot(descentePression_1mm['Temps (min)'],descentePression_1mm['P_LV (mbar)'],'k.', label= '1 mm')
    plt.legend()

    plt.rc('text', usetex=True)
    plt.rc('font', family='serif')
    plt.grid(b = True, which = 'major', axis = 'both')

    plt.errorbar(descentePression_sansCapots['Temps (min)'],descentePression_sansCapots['P_LV (mbar)'], yerr = descentePression_sansCapots['erreur P_LV '], fmt='none', ecolor = 'k', elinewidth = 1, capsize = 1, label= 'sans capots')
    plt.errorbar(descentePression_10mm['Temps (min)'],descentePression_10mm['P_LV (mbar)'], yerr = descentePression_10mm['erreur P_LV '], fmt='none', ecolor = 'k', elinewidth = 1, capsize = 1, label= '10 mm')
    plt.errorbar(descentePression_1mm['Temps (min)'],descentePression_1mm['P_LV (mbar)'], yerr = descentePression_1mm['erreur P_LV '], fmt='none', ecolor = 'k', elinewidth = 1, capsize = 1, label= '1 mm')

    plt.rc('text', usetex=True)
    plt.rc('font', family='serif')
    
    plt.yscale('log')
    plt.ylabel(r'$P_{LV}$ (mbar)')
    plt.xlim((0,1e2))
    plt.xscale('symlog', linthreshy=1e-1)#symlog displays a linear interval at 0 neighborhood
    xaxis = plt.gca().xaxis
    xaxis.set_minor_locator(MinorSymLogLocator(1e-1))
    plt.xlabel('Temps (min)')
    #plt.axes().yaxis.set_tick_params(which='minor', right = 'off')
    #plt.tick_params(axis='y', which='minor')
    #plt.ylim()
    #plt.minorticks_on()
    plt.grid(b = True, which = 'major', axis = 'both')
    
    #export en png et pgf pour latex.
    name = "pompage_pLv"
    plt.savefig(('Graphes/'+name+'.png'), format = 'png', transparent=True)
    plt.savefig(('Graphes/'+name+'.pgf'), format = 'pgf')
    plt.savefig(('Graphes/'+name+'.pdf'), format = 'pdf', transparent=True)
    plt.savefig(('Graphes/'+name+'.svg'), format = 'svg', transparent=True)
    plt.show()
#%% P_four et en fonction du temps lors du pompage
#Les deux figures précédentes en haut et en bas
if choix.Subplots:
    plt.figure(num='pFour', figsize = (largeurPlot, hauteurPlot))
    plt.subplot(2,1,1)
    plt.plot(descentePression_sansCapots['Temps (min)'],descentePression_sansCapots['P_LV (mbar)'],'r.', label= 'sans capots')
    plt.plot(descentePression_10mm['Temps (min)'],descentePression_10mm['P_LV (mbar)'],'b.', label= '10 mm')
    plt.plot(descentePression_1mm['Temps (min)'],descentePression_1mm['P_LV (mbar)'],'k.', label= '1 mm')
    plt.legend()

    plt.rc('text', usetex=True)
    plt.rc('font', family='serif')
    #plt.axes().yaxis.set_tick_params(which='minor', right = 'off')
    #plt.tick_params(axis='y', which='minor')
    #plt.xlim(); plt.ylim()
    #plt.minorticks_on()
    plt.grid(b = True, which = 'major', axis = 'both')

    plt.errorbar(descentePression_sansCapots['Temps (min)'],descentePression_sansCapots['P_LV (mbar)'], yerr = descentePression_sansCapots['erreur P_LV '], fmt='none', ecolor = 'k', elinewidth = 1, capsize = 1, label= 'sans capots')
    plt.errorbar(descentePression_10mm['Temps (min)'],descentePression_10mm['P_LV (mbar)'], yerr = descentePression_10mm['erreur P_LV '], fmt='none', ecolor = 'k', elinewidth = 1, capsize = 1, label= '10 mm')
    plt.errorbar(descentePression_1mm['Temps (min)'],descentePression_1mm['P_LV (mbar)'], yerr = descentePression_1mm['erreur P_LV '], fmt='none', ecolor = 'k', elinewidth = 1, capsize = 1, label= '1 mm')

    plt.rc('text', usetex=True)
    plt.rc('font', family='serif')

    plt.yscale('log')
    plt.ylabel('$P_{LV}$ (mbar)')
    plt.xlim((0,1e2))
    plt.xscale('symlog', linthreshy=1e-1)#symlog displays a linear interval at 0 neighborhood
    xaxis = plt.gca().xaxis
    xaxis.set_minor_locator(MinorSymLogLocator(1e-1))
    plt.title('Pompage primaire')
    plt.grid(b = True, which = 'major', axis = 'both')

    plt.subplot(2,1,2)
    plt.plot(descentePression_sansCapots['Temps (min)'],descentePression_sansCapots['P_HV (mbar)'],'r.', label= 'sans capots')
    plt.plot(descentePression_10mm['Temps (min)'],descentePression_10mm['P_HV (mbar)'],'b.', label= '10 mm')
    plt.plot(descentePression_1mm['Temps (min)'],descentePression_1mm['P_HV (mbar)'],'k.', label= '1 mm')
    plt.legend()

    plt.rc('text', usetex=True)
    plt.rc('font', family='serif')
    plt.grid(b = True, which = 'major', axis = 'both')

    plt.errorbar(descentePression_sansCapots['Temps (min)'],descentePression_sansCapots['P_HV (mbar)'], yerr = descentePression_sansCapots['erreur P_HV'], fmt='none', ecolor = 'k', elinewidth = 1, capsize = 1, label= 'sans capots')
    plt.errorbar(descentePression_10mm['Temps (min)'],descentePression_10mm['P_HV (mbar)'], yerr = descentePression_10mm['erreur P_HV'], fmt='none', ecolor = 'k', elinewidth = 1, capsize = 1, label= '10 mm')
    plt.errorbar(descentePression_1mm['Temps (min)'],descentePression_1mm['P_HV (mbar)'], yerr = descentePression_1mm['erreur P_HV'], fmt='none', ecolor = 'k', elinewidth = 1, capsize = 1, label= '1 mm')

    plt.rc('text', usetex=True)
    plt.rc('font', family='serif')

    plt.yscale('log')
    plt.ylabel(r'$P_{HV}$ (mbar)')
    plt.xscale('symlog', linthreshy=1e-1)#symlog displays a linear interval at 0 neighborhood
    xaxis = plt.gca().xaxis#minor ticks on symlog
    xaxis.set_minor_locator(MinorSymLogLocator(1e-1))
    plt.xlabel('Temps (min)')
    plt.title('Pompage secondaire')
    plt.grid(b = True, which = 'major', axis = 'both')
    plt.tight_layout()#pour séparer un peu les plots

    #export en png et pgf pour latex.
    name = "pompage_pFour"
    plt.savefig(('Graphes/'+name+'.png'), format = 'png', transparent=True)
    plt.savefig(('Graphes/'+name+'.pgf'), format = 'pgf')
    plt.savefig(('Graphes/'+name+'.pdf'), format = 'pdf', transparent=True)
    plt.savefig(('Graphes/'+name+'.svg'), format = 'svg', transparent=True)
    #plt.savefig(('Graphes/'+name+'.ps'), format = 'ps', transparent=True)
    plt.show()
#%% Des tests
#import numpy as np
#import matplotlib.pyplot as plt 
#plt.figure(figsize = (6.69, 5.1461538461538465))

# do some plotting here
#x = np.linspace(-2, 2, 100)
#plt.plot(x, x)

# save to file
#plt.savefig('test.pgf')